actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,487 @@
+                )
+            else:
+                # else no path to save
+                save_path = None
+                # diagnostic_sheet_save_path = (
+                #     diagnostic_sheet_dir / f"{metadata}_{row[str(metadata)]}.png"
+                # )
+
+            # Check skip
+            if not overwrite and save_path.is_file():
+                log.info(
+                    f"Skipping diagnostic sheet path for cell ID: {row.CellId},"
+                    f"{metadata} {row[str(metadata)]}"
+                )
+                return DiagnosticSheetResult(row.CellId, None)
+
+            # Return ready to save image
+            return DiagnosticSheetResult(row.CellId, str(save_path))
+        # Catch and return error
+        except Exception as e:
+            log.info(
+                f"Failed to retrieve the CellImage2DAllProjectionsPath"
+                f"for cell ID: {row.CellId},"
+                f"{metadata} {row[str(metadata)]}"
+                f"Error: {e}"
+            )
+            return DiagnosticSheetError(row.CellId, str(e))
+
+    @log_run_params
+    def run(
+        self,
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame],
+        max_cells: int = 1000,
+        distributed_executor_address: Optional[str] = None,
+        batch_size: Optional[int] = None,
+        overwrite: bool = False,
+        feature: Optional[str] = None,
+        metadata: Optional[Union[list, str]] = None,
+        fig_width: Optional[int] = None,
+        fig_height: Optional[int] = None,
+        **kwargs,
+    ):
+        """
+        Provided a dataset of single cell all projection images, generate a diagnostic
+        sheet grouped by desired metadata and feature
+
+        Parameters
+        ----------
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame]
+            The primary cell dataset to use for generating 
+            diagnistic sheet for a group of cells.
+
+            **Required dataset columns:** *["CellId"]*
+
+        batch_size: Optional[int]
+            An optional batch size to process n features at a time.
+            Default: None (Process all at once)
+
+        metadata: Union[str, list, None]
+            The metadata to group cells and generate a diagnostic sheet. 
+            For example, "FOVId" or "["FOVId", "ProteinDisplayName"]"
+
+        feature: str
+            The name of the single cell feature to display. For example, "imsize_orig"
+
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+            Default: None
+
+        clean: bool
+            Should the local staging directory be cleaned prior to this run.
+            Default: False (Do not clean)
+
+        debug: bool
+            A debug flag for the developer to use to manipulate how much data runs,
+            how it is processed, etc.
+            Default: False (Do not debug)
```

Comment:
I believe you are missing parameters in this docstring so last change is can you add the missing parameters and also making the docstring parameter order match the actual parameter order in the function definition? `batch_size` for example should be below `distributed_executor_address`